### PR TITLE
Topic/v140 clang c2

### PR DESCRIPTION
**Pull request by lovemycatt**
A few minor patches to enable compilation on Windows MSVC with v140_clang_c2 platform (currently clang 3.8).
A MSVC_v140_clang_c2.cmake has been added under cmake/. 
Some minor changes applied to CMakeLists.txt to enable install of .lib on Windows.

### CODE REVIEW

File: cmake/MSVC_v140_clang_c2.cmake [hunk 1]
```diff
@@ -0,0 +1,5 @@
+## FLAGS FOR MSVC v140_clang_c2
+set(CMAKE_CXX_FLAGS_DEBUG_INIT 			"/D_DEBUG /MTd /GR -Od -g2 -gdwarf-2 -std=c++1z -fms-extensions -fexceptions")
```
Comment (Neverlord) [inline cmake/MSVC_v140_clang_c2.cmake]: We don't use tabs.

File: libcaf_core/CMakeLists.txt [hunk 2]
```diff
@@ -129,7 +129,8 @@ if (NOT CAF_BUILD_STATIC_ONLY)
                         OUTPUT_NAME caf_core)
   if (CYGWIN)
     install(TARGETS libcaf_core_shared RUNTIME DESTINATION bin)
-  elseif (NOT WIN32)
+  else()
+# elseif (NOT WIN32)
```
Comment (Neverlord) [inline libcaf_core/CMakeLists.txt]: Please remove lines instead of commenting them out. There's no reason to keep them around.

File: libcaf_core/caf/config.hpp [hunk 3]
```diff
@@ -188,6 +188,12 @@
 #  define CAF_POSIX
 #endif
 
+#if defined (CAF_WINDOWS)
+#   if defined (__clang__)
+        struct IUnknown;    // fix for issue with static_cast<> in objbase.h (see https://github.com/philsquared/Catch/issues/690)
```
Comment (Neverlord) [inline libcaf_core/caf/config.hpp]: Coding style: we use 2 spaces, not 4.

File: libcaf_io/CMakeLists.txt [hunk 4]
```diff
@@ -43,7 +43,8 @@ if (NOT CAF_BUILD_STATIC_ONLY)
                         OUTPUT_NAME caf_io)
   if (CYGWIN)
     install(TARGETS libcaf_io_shared RUNTIME DESTINATION bin)
-  elseif (NOT WIN32)
+  else()
+#  elseif (NOT WIN32)
```
Comment (Neverlord) [inline libcaf_io/CMakeLists.txt]: Please also clean up commented-out lines in this file.

File: libcaf_io/caf/io/middleman.hpp [hunk 5]
```diff
@@ -43,7 +43,7 @@ namespace io {
 /// Manages brokers and network backends.
 class middleman : public actor_system::module {
 public:
-  friend class actor_system;
+  friend class ::caf::actor_system;
```
Comment (Neverlord) [inline libcaf_io/caf/io/middleman.hpp]: Is this change actually needed? `actor_system` is a known type at this point and there should not be any ambiguity.

Comment (lovemycatt) [reply libcaf_io/caf/io/middleman.hpp]: It actually generates a lot of the following tedious warnings.
It may be a bug of the specific clang compile version used by MS (it is a clang 3.8.0)

> 1> ...\actor-framework\libcaf_io\caf/io/middleman.hpp(46,27): warning : unqualified friend declaration referring to type outside of the nearest enclosing namespace is a Microsoft extension; add a nested name specifier [-Wmicrosoft-unqualified-friend]
> 1>    friend class actor_system;
> 1>                       ^
> 1>                       ::caf::
> 1>  1 warning generated.

If you think that such change is dirtier than the problem it wants to solve, I can  squash the other commits ad skip this.

Comment (Neverlord) [reply libcaf_io/caf/io/middleman.hpp]: If it generates warnings on your platform than it's ok. I just wondered what's the deal here.

**Comment by lovemycatt** [comment]
PR closed before resubmitting with suggested corrections.
